packages/cpp/ArmoniK.Api.Client/source/channel/ChannelFactory.cpp

```diff
@@ -0,0 +1,165 @@
+#include "channel/ChannelFactory.h"
+
+#include "exceptions/ArmoniKApiException.h"
+#include "options/ControlPlane.h"
+#include "utils/ChannelArguments.h"
+#include <grpcpp/create_channel.h>
+#include <grpcpp/security/credentials.h>
+#include <grpcpp/security/tls_credentials_options.h>
+
+#include <fstream>
+#include <sstream>
+#include <utility>
+
+namespace armonik {
+namespace api {
+namespace client {
+
+using namespace grpc::experimental;
+
+const std::string root_self_signed = R"(-----BEGIN CERTIFICATE REQUEST-----

```

Comment:
I agree it needs documentation. If I recall correctly, disabling the ssl validation still requires to pass a root cert. So this is just a dummy certificate to pass to the function.
It would also probably need the command used to generate it in case it needs to be recreated for whatever reason.